File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution(object):
    def generate(self, numRows):
        answer = []
        
        answer.append([1])        
        for i in range(1, numRows):
            sub = []
            prev = answer[i - 1]
            
            sub.append(1)
            for j in range(0, len(prev) - 1):
                n = prev[j] + prev[j + 1]
                sub.append(n)
                
            sub.append(1)
            answer.append(sub)
        
        return answer

File: Easy/0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> answer = new ArrayList();
        
        answer.add(List.of(1));
        
        for(int i = 1; i < numRows; i++)
        {
            List<Integer> subList = new ArrayList();            
            List<Integer> prevList = answer.get(i - 1);
            
            subList.add(1);
            for(int j = 0; j < prevList.size() - 1; j++)
            {
                int n = prevList.get(j) + prevList.get(j + 1);
                subList.add(n);
            }
            subList.add(1);
            
            answer.add(subList);
        }
        
        return answer;
    }
    
    
}

// iterate until numRows = 0

// get the last list of the list

// iterate the list

// set the first and the last index as 1

//  0, 1, 2, 3
//0   1, 2, 3  4


// n = last index n + n - 1

// 